src/lib/coaler/embedder/ConformerEmbedder.cpp

```diff
@@ -89,4 +137,118 @@ namespace coaler::embedder {
         params.clearConfs = false;
         return params;
     }
+
+    /*----------------------------------------------------------------------------------------------------------------*/
+
+    std::vector<multialign::PoseID> ConformerEmbedder::generateNewPosesForAssemblyLigand(
+        RDKit::ROMol *worstLigandMol, const multialign::LigandVector &targets,
+        const std::unordered_map<multialign::LigandID, multialign::PoseID> &conformerIDs) {
+        std::vector<unsigned> newIds;
+        for (const multialign::Ligand &target : targets) {
+            // find mcs
+            const multialign::LigandID targetID = target.getID();
+            const multialign::PoseID targetConformerID = conformerIDs.at(targetID);
+            const RDKit::ROMol targetMol = target.getMolecule();
+            const RDKit::Conformer &targetConformer = targetMol.getConformer(targetConformerID);
+
+            std::vector<RDKit::ROMOL_SPTR> mols = {
+                boost::make_shared<RDKit::ROMol>(*worstLigandMol),
+                boost::make_shared<RDKit::ROMol>(target.getMolecule()),
+            };
+
+            auto mcsResult = get_mcs(mols, false);
+            if (mcsResult.QueryMol == nullptr) {
+                spdlog::warn("no mcs found");
+                continue;
+            }
+
+            RDKit::SubstructMatchParameters substructMatchParams;
+            substructMatchParams.uniquify = true;
+            substructMatchParams.useChirality = false;
+            substructMatchParams.useQueryQueryMatches = false;
+            substructMatchParams.maxMatches = 1;
+            substructMatchParams.numThreads = 1;
+
+            std::vector<RDKit::MatchVectType> ligandMatches;
+            RDKit::SubstructMatch(*worstLigandMol, *mcsResult.QueryMol, ligandMatches, substructMatchParams.uniquify,
+                                  false, substructMatchParams.useChirality, substructMatchParams.useQueryQueryMatches,
+                                  substructMatchParams.maxMatches, substructMatchParams.numThreads);
+
+            std::vector<RDKit::MatchVectType> targetMatches;
+            RDKit::SubstructMatch(targetMol, *mcsResult.QueryMol, targetMatches, substructMatchParams.uniquify, false,
+                                  substructMatchParams.useChirality, substructMatchParams.useQueryQueryMatches,
+                                  substructMatchParams.maxMatches, substructMatchParams.numThreads);
+
+            if (targetMatches.empty() || ligandMatches.empty()) {
```

Comment:
the MCS should be found in the molecules. If you get no match this can be following the following cases afaik:
 - The molecules have wrong aromatizity -> you could try sanitizing both (usually this helps)
 - you substruct match params differ from the info the MCS uses -> you should try to fix those instead of altering the MCS imo